js/adapt-assessmentArticleModel.js

```diff
@@ -379,15 +375,93 @@ define([
             this.set("_isPass", isPass);
         },
 
-        _isAttemptsLeft: function() {
+        _overrideMarkingSettings: function() {
+          var questionConfig = this.getConfig()._questions;
+
+          var newMarkingSettings = {};
+
+          if (this._isMarkingSuppressionEnabled() && this._isAttemptsEnabled() && this.get('_attemptsLeft') > 0) {
+            this._suppressMarking();
+          } else {
+            this._unsuppressMarking();
```

Comment:
I think this would be tidier if it were more like:
```
var suppressMarking = (this._isMarkingSuppressionEnabled() && this._isAttemptsEnabled() && this.get('_attemptsLeft') > 0);

var newMarkingSettings = this._getMarkingSettings(suppressMarking);

for (var i = 0, l = this._currentQuestionComponents.length; i < l; i++) {
    this._currentQuestionComponents[i].set(newMarkingSettings, { pluginName: "_assessment" });
}

return newMarkingSettings;
```

That way you wouldn't need `_suppressMarking` or `_unsuppressMarking` - just `_getMarkingSettings`